mlflow/models/model.py

```diff
@@ -716,7 +716,11 @@ def log(
                     except Exception as e:
                         _logger.warning("Failed to load model config from %s: %s", model_config, e)
                 try:
-                    mlflow.tracking.fluent.log_params(model_config or {}, run_id=run_id)
+                    from mlflow.models.utils import _flatten_nested_params
+
+                    mlflow.tracking.fluent.log_params(
+                        _flatten_nested_params(model_config) or {}, run_id=run_id
+                    )
```

Comment:
i think this error handling seems okay, but want to call an alernative -- we can throw with a special message that we are unable to flatten params. advantage is that users would still get their nested params even if we aren't able to flatten them.
```
try:
    from mlflow.models.utils import _flatten_nested_params
    flattened_params = _flatten_nested_params(model_config)
except Exception as e:
    _logger.warning("Failed to flatten nested params: %s", str(e))

try:
    mlflow.tracking.fluent.log_params(
       flattened_params or {}, run_id=run_id
    )
except Exception as e:
    _logger.warning("Failed to log model config as params: %s", str(e))
```